src/typechecker/FStar.TypeChecker.Tc.fst

```diff
@@ -893,15 +907,18 @@ let tc_decl env se: list sigelt * list sigelt * Env.env =
      BU.print1 "Processing %s\n" (U.lids_of_sigelt se |> List.map string_of_lid |> String.concat ", ");
    if Env.debug env Options.Low then
      BU.print1 ">>>>>>>>>>>>>>tc_decl %s\n" (Print.sigelt_to_string se);
-   if se.sigmeta.sigmeta_admit
-   then begin
-     let old = Options.admit_smt_queries () in
-     Options.set_admit_smt_queries true;
-     let result = tc_decl' env se in
-     Options.set_admit_smt_queries old;
-     result
-   end
-   else tc_decl' env se
+   let tc_ses, new_ses, env'
+     = if se.sigmeta.sigmeta_admit
+       then begin
+         let old = Options.admit_smt_queries () in
+         Options.set_admit_smt_queries true;
+         let result = tc_decl' env se in
+         Options.set_admit_smt_queries old;
+         result
+       end
+       else tc_decl' env se in
+   // Typecheck attributes
+   List.map (tc_decl_attributes env) tc_ses, new_ses, env'
```

Comment:
Shouldn't this check be done on `se` on entry to this function? Or in `tc_decl'`.

Reply:
Right, that's way cleaner indeed, just pushed a commit moving it to `tc_decl'`.